confluent_kafka/schema_registry/json_schema.py

```diff
@@ -0,0 +1,300 @@
+        if not callable(self._subject_name_func):
+            raise ValueError("subject.name.strategy must be callable")
+
+        if len(conf_copy) > 0:
+            raise ValueError("Unrecognized properties: {}"
+                             .format(conf_copy.keys()))
+
+        schema_dict = json.loads(schema_str)
+        schema_name = schema_dict.get('title', None)
+        if schema_name is None:
+            raise ValueError("Missing required JSON schema annotation title")
+
+        self._schema_name = schema_name
+        self._parsed_schema = schema_dict
+        self._schema = Schema(schema_str, schema_type="JSON")
+
+    def __call__(self, ctx, obj):
+        """
+        Serializes an object to the Confluent Schema Registry's JSON binary
+        format.
+
+        Args:
+            ctx (SerializationContext): Metadata pertaining to the serialization
+                operation.
+
+            obj (object): object instance to serialize.
+
+
+        Note:
+            None objects are represented as Kafka Null.
+
+        Raises:
+            SerializerError if any error occurs serializing obj
+
+        Returns:
+            bytes: Confluent Schema Registry formatted JSON bytes
+
+        """
+        if obj is None:
+            return None
+
+        subject = self._subject_name_func(ctx, self._schema_name)
+
+        # Check to ensure this schema has been registered under subject_name.
+        if self._auto_register and subject not in self._known_subjects:
+            # The schema name will always be the same. We can't however register
+            # a schema without a subject so we set the schema_id here to handle
+            # the initial registration.
+            self._schema_id = self._registry.register_schema(subject,
+                                                             self._schema)
+            self._known_subjects.add(subject)
+        elif not self._auto_register and subject not in self._known_subjects:
+            registered_schema = self._registry.lookup_schema(subject,
+                                                             self._schema)
+            self._schema_id = registered_schema.schema_id
+            self._known_subjects.add(subject)
+
+        if self._to_dict is not None:
+            value = self._to_dict(ctx, obj)
+        else:
+            value = obj
+
+        try:
+            validate(instance=value, schema=self._parsed_schema)
+        except ValidationError as ve:
+            raise SerializationError(ve.message)
+
+        with _ContextStringIO() as fo:
+            # Write the magic byte and schema ID in network byte order (big endian)
+            fo.write(struct.pack('>bI', _MAGIC_BYTE, self._schema_id))
+            # JSON dump always writes a str never bytes
+            # https://docs.python.org/3/library/json.html
+            fo.write(json.dumps(value).encode('utf8'))
+
+            return fo.getvalue()
+
+
+class JsonDeserializer(Deserializer):
```

Comment:
Should this perhaps be JSONDeserializer?
See json.JSONDecoder

Reply:
It was originally, I disliked the way it looked. I don't have strong feelings though so I'll let @mhowlett break the tie.

Reply:
what @magnus says. from PEP-8:

> Note: When using acronyms in CapWords, capitalize all the letters of the acronym. Thus HTTPServerError is better than HttpServerError.